### PR TITLE
Mass update of dependencies

### PR DESCRIPTION
**Pull request by tjormola**
I wanted to build a new feature for json-avro-converter (PR will follow soon). I was developing with JDK9 and noticed some of the Gradle plugins weren't compatible with the defined versions. So I bumped all the dependencies to the latest versions as of 01/28/2018 and was then able to build and run the tests without my modifications. Here's the new dependency updates, please apply if you like it. Of course I couldn't test the Sonar and release related stuff, hopefully those still work for your workflow...

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
+#Wed Feb 28 18:15:02 EET 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.2.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.5.1-all.zip
```
Comment (adamdubiel) [inline gradle/wrapper/gradle-wrapper.properties]: can we switch back to -bin? all is unnecessary overhead in CI builds

File: build.gradle [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+    }
+}
+
 plugins {
     id 'java'
-    id 'pl.allegro.tech.build.axion-release' version '1.8.1'
+    id 'pl.allegro.tech.build.axion-release' version '1.9.0'
 }
 
+apply plugin: 'io.codearte.nexus-staging'
```
Comment (adamdubiel) [inline build.gradle]: why did you move this plugin to root? we only wanted to publish `converter`, do you need `validator` as well?

**Comment by adamdubiel** [comment]
I misclicked approve, i would like you to answer/fix my comments.

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (tjormola) [reply gradle/wrapper/gradle-wrapper.properties]: Sorry about that. I just blindly did a gradle dist without specifying --distribution-type. Will fix.

File: build.gradle (hunk 2 above)
Comment (tjormola) [reply build.gradle]: Even with the currently used 0.5.3 version the nexus staging plugin emits the warning

`WARNING. Nexus staging plugin should only be applied to the root project in build.`

When updated to 0.11.0 it won't allow anything but applying the plugin from the root project, as documented at https://github.com/Codearte/gradle-nexus-staging-plugin#multi-project-build.

But I really haven't used this plugin and I definitely shouldn't be messing around with your release workflow, I will just revert updating this plugin in the rebased PR since the project builds fine for me without updating of this plugin. You do what ever you wish with it or do nothing ;)

Comment (tjormola) [reply build.gradle]: Updated in #18. Please discard this.

**Comment by tjormola** [comment]
Closing this myself ;)

File: build.gradle (hunk 2 above)
Comment (adamdubiel) [reply build.gradle]: I didn't know it changed so much :) I will have a look at it, but i would rather do it myself, otherwise you would have to wait a long time before we release :D
